src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -17,67 +17,128 @@
 import org.openqa.selenium.StaleElementReferenceException;
 import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.ArrayList;
-
 /**
- * Represents CdfPipelineRunAction
+ * Represents Cdf Pipeline Run Page Actions
  */
 public class CdfPipelineRunAction {
   private static final Logger logger = LoggerFactory.getLogger(CdfPipelineRunAction.class);
-  public static CdfPipelineRunLocators cdfPipelineRunLocators;
 
   static {
-    cdfPipelineRunLocators = SeleniumHelper.getPropertiesLocators(CdfPipelineRunLocators.class);
+    SeleniumHelper.getPropertiesLocators(CdfPipelineRunLocators.class);
+  }
+
+  /**
+   * Click on the Run button
+   */
+  public static void runClick() {
+    ElementHelper.clickOnElement(CdfPipelineRunLocators.run);
   }
 
-  public static void runClick() throws InterruptedException {
-    ElementHelper.clickOnElement(cdfPipelineRunLocators.run);
+  /**
+   * Click on the Run button inside 'Runtime Arguments' dialog
+   */
+  public static void clickDeployedConfigRunButton() {
+    ElementHelper.clickOnElement(CdfPipelineRunLocators.deployedConfigRunButton);
   }
 
+  /**
+   * Get the Pipeline's Status bubble color
+   *
+   * @return color attribute value
+   */
   public static String runPipelineStatus() {
-    return cdfPipelineRunLocators.runPipelineStatus.getAttribute(ConstantsUtil.COLOR);
+    return CdfPipelineRunLocators.runPipelineStatus.getAttribute(ConstantsUtil.COLOR);
   }
 
-  public static Boolean isRunning() {
-    return provideState(cdfPipelineRunLocators.runningStatus);
+  /**
+   * Check if the Pipeline is in the expected Status
+   *
+   * @param element WebElement of Pipeline's expected Status
+   * @return boolean
+   */
+  private static boolean provideState(WebElement element) {
+    return ElementHelper.isElementDisplayed(element);
   }
 
-  public static void logsClick() {
-    cdfPipelineRunLocators.logs.click();
+  /**
+   * Check if the pipeline is in Running Status
+   *
+   * @return Boolean
+   */
+  public static Boolean isRunning() {
+    return provideState(CdfPipelineRunLocators.runningStatus);
   }
 
+  /**
+   * Check if the pipeline is in Provisioning Status
+   *
+   * @return Boolean
+   */
   public static Boolean isProvisioning() {
-    return provideState(cdfPipelineRunLocators.provisioningStatus);
+    return provideState(CdfPipelineRunLocators.provisioningStatus);
   }
 
-  private static boolean provideState(WebElement element) {
-    boolean bool = false;
-    try {
-      bool = element.isDisplayed();
-    } catch (NoSuchElementException e) {
-      logger.error("Element not found" + e);
-    }
-    return bool;
+  /**
+   * Wait till the Pipeline's status changes (from Running) to either Succeeded or Failed within the
+   * Timeout: {@link ConstantsUtil#PIPELINE_RUN_TIMEOUT_SECONDS}
+   */
+  public static void waitTillPipelineRunCompletes() {
```

Comment:
Include `stopped` as well in completed state.

Reply:
Added.